modules/microshift-oc-mirror-transform-imageset-to-crio.adoc

```diff
@@ -0,0 +1,94 @@
+//Module included in the following assemblies:
+//
+//* microshift_running_apps/microshift-operators-olm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-oc-mirror-transform-imageset-to-crio_{context}"]
+= Configuring CRI-O for using registry mirror for Operators
+
+Use the following example procedure to transform the `imageContentSourcePolicy.yaml` created with the oc mirror tool into a setting you can add to the CRI-O container runtime configuration.
+
+.Prerequisites
+* The {oc-first} is installed.
+* Operator Lifecycle Manager (OLM) is installed.
+* The oc-mirror OpenShift CLI (oc) plugin is installed.
+* The yq binary is installed.
+* `ImageContentSourcePolicy` and `CatalogSource` YAML files are available in the `oc-mirror-workspace/results-*` directory.
+
+.Procedure
+
+. Confirm the contents of the `ImageContentSourcePolicy` YAML by running the following command:
++
+[source,terminal]
+----
+$ cat oc-mirror-workspace/<results-1707148826>/imageContentSourcePolicy.yaml <1>
+----
+<1> Replace _<results-1707148826>_ with the results directory name.
++
+.Example output
+[source,terminal]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  labels:
+    operators.openshift.org/catalog: "true"
+  name: operator-0
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - registry.<example.com>/amq7 <1>
+    source: registry.redhat.io/amq7
+----
+<1> Example name. Your registry name is listed here.
+
+. Transform the `imageContentSourcePolicy.yaml` into a format ready for CRI-O configuration by running the following command:
++
+[source,terminal]
+----
+yq '.spec.repositoryDigestMirrors[] as $item ireduce([]; . + [{"mirror": $item.mirrors[], "source": ($item | .source)}]) | .[] |
+  "[[registry]]
+      prefix = \"" + .source + "\"
+      location = \"" + .mirror + "\"
+      mirror-by-digest-only = true
+      insecure = true
+      "' ./icsp.yaml
+----
++
+.Example output
+[source,terminal]
+----
+[[registry]]
+      prefix = "registry.redhat.io/amq7"
+      location = "registry.example.com/amq7"
+      mirror-by-digest-only = true
+      insecure = true
+----
+
+. Add the output to the CRI-O configuration file in the `/etc/containers/registries.conf.d/` directory:
++
+.Example mirror configuration file
+[source,terminal]
+----
+[[registry]]
+      prefix = "registry.redhat.io/amq7"
+      location = "registry.example.com/amq7"
+      mirror-by-digest-only = true
+      insecure = true
+
+[[registry]]
+    prefix = ""
+    location = "quay.io"
+    mirror-by-digest-only = true
+[[registry.mirror]]
+    location = "<registry_host>:<port>" <1>
+    insecure = false
+----
+<1> Replace `<registry_host>:<port>` with the host name and port of your mirror registry server, for example, `<microshift-quay:8443>`.
+
+. Apply the CRI-O configuration changes by restarting {microshift-short} with the following command:
++
+[source,terminal]
+----
+$ sudo systemctl restart microshift
```

Comment:
@ShaunaDiaz actually no need to restart microshift, just restarting crio is ok.
```console
sudo systemctl restart crio
```

Reply:
@kuiwang02 I'm not actually certain that's true; we say in the docs to restart MicroShift if _any_ config changes are made. @pmtk Could you confirm which way we can go here?

Reply:
@kuiwang02  is right. In this case we want to restart or send SIGHUP to crio, not microshift